pgjdbc/src/main/java/org/postgresql/core/BaseConnection.java

```diff
@@ -64,19 +66,86 @@ ResultSet execSQLQuery(String s, int resultSetType, int resultSetConcurrency)
   ReplicationProtocol getReplicationProtocol();
 
   /**
-   * <p>Construct and return an appropriate object for the given type and value. This only considers
-   * the types registered via {@link org.postgresql.PGConnection#addDataType(String, Class)} and
-   * {@link org.postgresql.PGConnection#addDataType(String, String)}.</p>
+   * {@inheritDoc}
+   *
+   * <p>
+   * Performs a defensive copy, per specification that returned map should be
```

Comment:
I cannot see where the map needs to be modifiable in the spec ?

Reply:
Per https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#getTypeMap--

"You must invoke setTypeMap after making changes to the Map object returned from getTypeMap as a JDBC driver may create an internal copy of the Map object passed to setTypeMap"

This implies that the returned Map must be modifiable.

And the tutorial shows the use-case of it being always modifiable:
https://docs.oracle.com/javase/tutorial/jdbc/basics/sqlcustommapping.html#using_connection_type_map